spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,261 @@
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('ping pr author regarding merge conflict', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+
+      const link = 'link'.link(
+        'https://help.github.com/articles/resolving-a-merge' +
+          '-conflict-using-the-command-line/'
+      );
+      const params = {
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        number: payloadData.payload.pull_request.number,
+        body:
+          'Hi @' +
+          payloadData.payload.pull_request.user.login +
+          '. Due to recent changes in the "develop" branch, ' +
+          'this PR now has a merge conflict. ' +
+          'Please follow this ' +
+          link +
+          ' if you need help resolving the conflict, ' +
+          'so that the PR can be merged. Thanks!',
+      };
+      expect(github.issues.createComment).toHaveBeenCalledWith(params);
+    });
+
+    it('adds merge conflict label', () => {
+      expect(github.issues.addLabels).toHaveBeenCalled();
+      const params = {
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        number: payloadData.payload.pull_request.number,
+        labels: [mergeConflictLabel.name],
+      };
+      expect(github.issues.addLabels).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe('pull request merge does not result in merge conflict', () => {
+    beforeEach(async () => {
+      // Simulate a merged pull request.
+      payloadData.payload.pull_request.merged = true;
+      payloadData.payload.action = 'closed';
+
+      const pullRequest = { ...payloadData.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = true;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('checks for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('does not ping pr author', () => {
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+    it('does not add merge conflict label', () => {
+      expect(github.issues.addLabels).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('pull request already has merge conflict label', () => {
+    beforeEach(async () => {
+      // Simulate synchronize event.
```

Comment:
Detail this a bit more -- what this simulation exactly do.

Reply:
Done.